File: algorithms/grokking-algorithms/src/main/java/chapter1/BinarySearch.java
package chapter1;

public class BinarySearch {

  private final int[] sortedArray;

  public BinarySearch(int[] sortedArray) {
    this.sortedArray = sortedArray;
  }

  public int indexOf(int item) {
    int low = 0;
    int hig = sortedArray.length - 1;
    while (low <= hig) {
      int mid = (low + hig) / 2;
      if (sortedArray[mid] == item) {
        return mid;
      }
      if (sortedArray[mid] < item) {
        low = mid + 1;
      } else {
        hig = mid - 1;
      }
    }
    return -1;
  }
}

File: algorithms/grokking-algorithms/python/chapter1/BinarySearch.py
def binary_search(sorted_items: list, seeking_item: int):
    low = 0
    high = len(sorted_items) - 1
    while low <= high:
        mid = (low + high) // 2
        guess = sorted_items[mid]
        if guess == seeking_item:
            return mid
        if guess < seeking_item:
            low = mid + 1
        else:
            high = mid - 1
    return None
